src/solvers/dgsem_tree/dg_2d.jl

```diff
@@ -30,6 +30,10 @@ function create_cache(mesh::TreeMesh{2}, equations,
   cache = (;cache..., create_cache(mesh, equations, dg.volume_integral, dg, uEltype)...)
   cache = (;cache..., create_cache(mesh, equations, dg.mortar, uEltype)...)
 
+  # TODO: Just defined for VolumeIntegralShockCapturingSubcell.
+  # It adds flux_antidiffusive, which should save data for alle element
+  cache = (;cache..., create_cache(cache, equations, dg, uEltype)...)
```

Comment:
I don't understand what you mean. I think this should be handled by `cache = (;cache..., create_cache(mesh, equations, dg.volume_integral, dg, uEltype)...)` above? You don't seem to use the

> the number of elements

in the method you implemented below? Could you please explain the purpose and problems in more details?

Reply:
First, thanks for all the answers.
In this new function [`create_mesh`](https://github.com/trixi-framework/Trixi.jl/pull/1036/files#diff-0795950371176d2c476c8fc9454249d4acb60bafc9f2070601cc0b5f256d47b2R65-R72) in `methods_SSP` (maybe not the best place for it), I add the antidiffusive flux to the cache. For this, I need the existing `cache` to use `nelements(dg, cache)` or the defined `elements` from the `cache` for `nelements(elements)`. These are not given in the existing function you mentioned.

Am I missing a way to calculate `nelement` without `cache`? Or alternatively, should I just add cache as a parameter to the existing function like `cache = (;cache..., create_cache(cache, mesh, equations, dg.volume_integral, dg, uEltype)...)`?

Reply:
When you use AMR, the number of elements will change anyways. Thus, you need some checks and `resize!` stuff inside the function where you use containers of the size of all elements, see for example 
https://github.com/trixi-framework/Trixi.jl/blob/53406c4d8a26e835d96c7e2fe575d51ea2f3f9be/src/solvers/dgsem_tree/indicators_1d.jl#L32-L38

Reply:
Yes, that's true. So, I should initialize the antidiffusive flux here like you suggested for the alphas with `Array{uEltype, 4}` and resize it in the right functions?

Reply:
Basically yes. However, you can't `resize!` multi-dimensional arrays in Julia. Thus, you need to use a plain `Vector` as buffer, `resize!` that if necessary, and wrap it in a multi-dimensional array. See our container infrastructure I linked in https://github.com/trixi-framework/Trixi.jl/pull/1036#discussion_r807912809 below.

Reply:
I just looked at it again and realized my mistake. Thanks.

Reply:
I still have some problems with that.
If I understand this correctly, the `ElementContainer` is initialized in
> https://github.com/trixi-framework/Trixi.jl/blob/53406c4d8a26e835d96c7e2fe575d51ea2f3f9be/src/solvers/dgsem_tree/containers_1d.jl#L50-L70

Here, the ingredients already get a size using the `capacity`/`nelements`, which is already known. Later, the size is just changed accordingly due to AMR effects in `Base.resize!()`.

When I create my data in `create_cache`, the starting number of elements is not known in the function which is called directly from 

> https://github.com/bennibolm/Trixi.jl/blob/957d8283bdb2fcdc5861caf39a1e436ab84bc77a/src/solvers/dgsem_tree/indicators.jl#L175-L184

I can create the undefined Vector with `Vector{real(basis)}()`. Is there any equivalent command to create an Array with no defined size? I couldn't find one.

Reply:
Just set the size to zero elements. When you call the function using the stuff the first time, the number of elements will not be zero and the buffer can be `resize!`ed accordingly. You may need to wrap this stuff in a kind of
```julia
mutable struct ContainerFCT{uEltype<:Real}
  antidiffusive_flux1::Array{uEltype, 4} # [variables, i, j, elements]
  antidiffusive_flux2::Array{uEltype, 4} # [variables, i, j, elements]
   # internal `resize!`able storage
  _antidiffusive_flux1::Vector{uEltype}
  _antidiffusive_flux2::Vector{uEltype}
end
```
implementing `resize!` and put that in the `cache` since the `cache` is an (immutable) `NamedTuple`.